src/NosCore.Controllers/Friend/MJoinPacketHandler.cs

```diff
@@ -45,20 +45,31 @@ public MJoinPacketHandler(IFriendHttpClient friendHttpClient, IMinilandProvider
         public override void Execute(MJoinPacket mJoinPacket, ClientSession session)
         {
             var target = Broadcaster.Instance.GetCharacter(s => s.VisualId == mJoinPacket.VisualId);
-            if ((target != null) && _friendHttpClient.GetListFriends(session.Character.CharacterId)
-                .Any(s => s.CharacterId == mJoinPacket.VisualId))
+            var friendList = _friendHttpClient.GetListFriends(session.Character.CharacterId);
+            if (target != null && friendList.Any(s => s.CharacterId == mJoinPacket.VisualId))
             {
-                var info = _minilandProvider.GetMiniland(mJoinPacket.VisualId);
-                if (info.State == MinilandState.Open)
+                var miniland = _minilandProvider.GetMiniland(mJoinPacket.VisualId);
+                if (miniland.State == MinilandState.Open)
                 {
-                    session.ChangeMapInstance(info.MapInstanceId, 5, 8);
+                    session.ChangeMapInstance(miniland.MapInstanceId, 5, 8);
                 }
                 else
                 {
+                    if (miniland.State == MinilandState.Private)
+                    {
+                        if (friendList.Where(w => w.RelationType != CharacterRelationType.Blocked)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Merge this if statement with the enclosing one.](https://app.codacy.com/app/NosCoreIO/NosCore/pullRequest?prid=4513002)